prow/plugins/reopen/reopen.go

```
@@ -74,9 +75,23 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent) e
 
 	if e.IsPR {
 		log.Info("Re-opening PR.")
-		return gc.ReopenPR(org, repo, number)
+		if err := gc.ReopenPR(org, repo, number); err != nil {
+			if scbc, ok := err.(github.StateCannotBeChanged); ok {
+				resp := fmt.Sprintf("failed to re-open PR: %s", scbc)
+				return gc.CreateComment(org, repo, number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, resp))
+			}
+			return err
+		}
+	} else {
+		log.Info("Re-opening issue.")
+		if err := gc.ReopenIssue(org, repo, number); err != nil {
+			if scbc, ok := err.(github.StateCannotBeChanged); ok {
+				resp := fmt.Sprintf("failed to re-open Issue: %s", scbc)
+				return gc.CreateComment(org, repo, number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, resp))
+			}
+			return err
+		}
```

er no, this should return the err or nil always

---

moved this back closer to the original code.